### PR TITLE
Use inbuilt icu

### PR DESCRIPTION
**Pull request by Tobias-Fischer**
- [x] OSX Build -- Tobias-Fischer
- [x] OSX ARM Build -- Tobias-Fischer
- [x] Windows Build
- [x] Linux Build -- hmaarrfk q server
- [x] Linux arm Build -- hmaarrfk cf server

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by Tobias-Fischer** [comment]
@conda-forge-admin, please rerender

**Comment by Tobias-Fischer** [comment]
See conversation in #6 for reasoning about this PR. 

In short: All previous qt-webengine builds used inbuilt icu (by mistake) already. However, they pulled in icu as a host dependency (although that system/host icu was not being used). In turn, this constrained the versions of `icu` that `qt-webengine` was compatible with - without any reason.

This PR removes the `icu` dependency so that the inbuilt `icu` will be used AND the build will be compatible with any conda `icu` at runtime.

Fixes #6, needed for https://github.com/conda-forge/qt-feedstock/pull/227 to get the latest qt-main and qt-webengine.

**Comment by Tobias-Fischer** [comment]
This PR would require a rebuild for all platforms.

**Comment by Tobias-Fischer** [comment]
This PR is ready for review @conda-forge/qt-webengine @andfoy @hmaarrfk

**Comment by hmaarrfk** [comment]
Ok, i added a delegation list above.

**Comment by hmaarrfk** [comment]
As a note to others, system provided ICU was never used in previous builds due to a missing compiler flag.

**Review by hmaarrfk — APPROVED** [review]

**Review by andfoy — APPROVED** [review]

**Comment by hmaarrfk** [comment]
[linux_aarch64.zip](https://github.com/conda-forge/qt-webengine-feedstock/files/8701847/linux_aarch64.zip)

**Comment by Tobias-Fischer** [comment]
osx-64 build logs:
[osx-64.zip](https://github.com/conda-forge/qt-webengine-feedstock/files/8704246/osx-64.zip)
 
build: https://anaconda.org/TobiasRobotics/qt-webengine/files (cfep03 label)

**Comment by Tobias-Fischer** [comment]
osx-arm64 build logs:

[osx-arm64.zip](https://github.com/conda-forge/qt-webengine-feedstock/files/8704250/osx-arm64.zip)

build: https://anaconda.org/TobiasRobotics/qt-webengine/files (cfep03 label)

**Comment by hmaarrfk** [comment]
I'm attempting to build windows now. it may take a while.

**Comment by hmaarrfk** [comment]
Windows build logs:
[win_64_log.zip](https://github.com/conda-forge/qt-webengine-feedstock/files/8707765/win_64_log.zip)

https://anaconda.org/mark.harfouche/qt-webengine/files

**Comment by hmaarrfk** [comment]
I guess i have to review logs and stuff. Will do tonight.

File: recipe/build.sh [hunk 1]
```diff
@@ -87,6 +87,11 @@ if [[ $(uname) == "Darwin" ]]; then
     if [[ $(arch) == "arm64" ]]; then
       EXTRA_FLAGS="QMAKE_APPLE_DEVICE_ARCHS=arm64"
     fi
+    
+    if [[ "${CONDA_BUILD_CROSS_COMPILATION:-}" == "1" ]]; then
+       # Get arm64 sysconfigdata for the python2_hack
+       cp ${SRC_DIR}/python2_hack/lib/python2.7/_sysconfigdata_x86_64_apple_darwin13_4_0.py ${SRC_DIR}/python2_hack/lib/python2.7/_sysconfigdata_arm64_apple_darwin20_0_0.py
+    fi
```
Comment (isuruf) [inline recipe/build.sh:94]: Instead of doing this, unset `_CONDA_PYTHON_SYSCONFIGDATA_NAME` env variable.

Comment (Tobias-Fischer) [reply recipe/build.sh]: Thanks, done. Rebuilding now for osx-arm64.

**Comment by Tobias-Fischer** [comment]
osx-arm64 build after @isuruf review comment logs: 
[osx-arm64-new.zip](https://github.com/conda-forge/qt-webengine-feedstock/files/8715128/osx-arm64-new.zip)


build (force overwritten the previous build as hash did not change): https://anaconda.org/TobiasRobotics/qt-webengine/files (cfep03 label)

**Comment by hmaarrfk** [comment]
Thanks everybody!

**Pull request merged**